### PR TITLE
Named Entity Recognition

### PR DESCRIPTION
**Pull request by LennoxHB**
Used huggingface to do NER in the scrapy pipeline, results are uploaded to mongo.
Managed to get mostly good results with a few unexpected ones.

image

### CODE REVIEW

File: scraper/domainScraper/pipelines/ner.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    
+    def process_item(self, item, spider):
+        item = DomainAnalyitcs(item)
+
+        tokenizer = AutoTokenizer.from_pretrained("dslim/bert-base-NER")
+        model = AutoModelForTokenClassification.from_pretrained("dslim/bert-base-NER")
+        nlp = pipeline("ner", model=model, tokenizer=tokenizer)
+
+        ner_results = nlp(item['raw'])
+        item['ner'] = []
```
Comment (ALADM2) [inline scraper/domainScraper/pipelines/ner.py:16]: Can you explain what's happening in this for loop?

**Review by ALADM2 — APPROVED** [review]
Looking good. That is interesting data for the front end. Have to think how to display it.

File: scraper/domainScraper/pipelines/ner.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        for result in ner_results:                            
+            if result['word'][0:2] == '##':
+                item['ner'][len(item['ner']) - 1] = item['ner'][len(item['ner']) - 1] + result['word'][2:]
+            elif result['entity'][0] == 'B':
+                item['ner'].append(result['word'])
+            elif result['entity'][0] == 'I':
+                item['ner'][len(item['ner']) - 1] = item['ner'][len(item['ner']) - 1] + " " + result['word']
```
Comment (ALADM2) [inline scraper/domainScraper/pipelines/ner.py:39]: Can you explain what's happening in this for loop?

**Review by ALADM2 — APPROVED** [review]
Looking good. That is interesting data for the front end. Have to think how to display it.

File: scraper/domainScraper/pipelines/ner.py (hunk 2 above)
Comment (LennoxHB) [reply scraper/domainScraper/pipelines/ner.py:39]: I've added some comments to the code to explain what is going on.

**Review by abdallah-altay — APPROVED** [review]
Nice work. The data is looking better with this.

**Pull request merged**